packages/diagrams/frontend/sirius-components-diagrams-reactflow/src/renderer/palette/DiagramElementPalette.tsx

```diff
@@ -11,17 +11,24 @@
  *     Obeo - initial API and implementation
  *******************************************************************************/
 
-import { memo } from 'react';
+import { memo, useContext } from 'react';
+import { DiagramContext } from '../../contexts/DiagramContext';
+import { DiagramContextValue } from '../../contexts/DiagramContext.types';
 import { useDiagramDirectEdit } from '../direct-edit/useDiagramDirectEdit';
 import { DiagramElementPaletteProps } from './DiagramElementPalette.types';
 import { Palette } from './Palette';
 import { PalettePortal } from './PalettePortal';
 import { useDiagramElementPalette } from './useDiagramElementPalette';
 
 export const DiagramElementPalette = memo(({ diagramElementId, labelId }: DiagramElementPaletteProps) => {
+  const { readOnly } = useContext<DiagramContextValue>(DiagramContext);
   const { isOpened, x, y } = useDiagramElementPalette();
   const { setCurrentlyEditedLabelId } = useDiagramDirectEdit();
 
+  if (readOnly) {
```

Comment:
Configure this instead in `useDiagramElementPalette` and `useDiagramPalette` because here you can have a state which says that the palette `isOpened` but no palette.

Reply:
Done